docs/source/integrations/plugins.md

```diff
@@ -447,7 +455,11 @@ executionDidStart?(
 
 ### `willResolveField`
 
-The `willResolveField` event fires whenever Apollo Server is about to resolve a single field during the execution of an operation. It is passed an object with four fields `source`, `args`, `context`, and `info`, corresponding to the [four positional arguments passed to resolvers](../data/resolvers/#resolver-arguments). (Note that the `source` field goes with the argument we often called `parent` in our docs.) The `willResolveField` handler is called on the object returned from your `executionDidStart` handler. It may return an ["end hook"](#end-hooks) function that is invoked with the result returned by your resolver or the error that it throws. The end hook is called once your resolver has fully resolved: for example, if it returns a Promise, the end hook is called with the Promise's eventual resolved result.
+The `willResolveField` event fires whenever Apollo Server is about to resolve a single field during the execution of an operation. The handler is passed an object with four fields (`source`, `args`, `context`, and `info`) that correspond to the [four positional arguments passed to resolvers](../data/resolvers/#resolver-arguments). (Note that `source` corresponds to the argument often called `parent` in these docs.)
+
+You provide your `willResolveField` handler in the object returned by your [`executionDidStart`](#executiondidstart) handler.
+
+Your `willResolveField` handler can optionally return an ["end hook"](#end-hooks) function that's invoked with the resolver's result (or the error that it throws). The end hook is called when your resolver has _fully_ resolved (i.e., if the resolver returns a Promise, the hook is called with the Promise's eventual resolved result).
```

Comment:
Probably more of an `e.g.` than an `i.e.` as there are other things like "returning a Promise resolving to an array of Promises resolving to lists of Promises"